amivapi/cron.py

```diff
@@ -114,7 +118,11 @@ def schedule_task(time, func, *args):
 
 def update_scheduled_task(time, func, *args):
     """ Update a scheduled task that was previously registered. """
-    func_s = func_str(func)
+    if func_str(func) is "remindermail":
+        item = pickle.load(args)
+        func_s = "remindermail_"+str(item['_id'])
```

Comment:
I think this will break the update. the `update_one` below identifies the function by name, so if the name is changed, the original function cannot be updated anymore.